src/aws-redshiftserverless/workgroup.ts

```diff
@@ -0,0 +1,351 @@
+ *
+ * const nameSpace = new Workgroup(
+ *   stack,
+ *   'Workgroup',
+ *   {
+ *     workgroupName: 'my-workgroup',
+ *     namespace: namespace,
+ *     vpc,
+ *   },
+ * );
+ */
+export class Workgroup extends Resource implements IWorkgroup {
+  /**
+   * Import an existing workgroup to the stack from its attributes.
+   */
+  public static fromWorkgroupAttributes(scope: Construct, id: string, attrs: WorkgroupAttributes): IWorkgroup {
+    class Import extends Resource implements IWorkgroup {
+      public readonly workgroupName = attrs.workgroupName;
+      public readonly workgroupId = attrs.workgroupId;
+      public readonly endpointAddress = attrs.endpointAddress;
+      public readonly port = attrs.port;
+      public readonly connections = new aws_ec2.Connections({
+        securityGroups: attrs.securityGroups,
+        defaultPort: aws_ec2.Port.tcp(attrs.port),
+      });
+      public readonly workgroupArn = Stack.of(this).formatArn({
+        resource: 'redshift-serverless',
+        service: 'workgroup',
+        resourceName: attrs.workgroupId,
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The workgroup Arn
+   */
+  readonly workgroupArn: string;
+
+  /**
+   * The workgroup name
+   */
+  readonly workgroupName: string;
+
+  /**
+   * The workgroup id
+   */
+  readonly workgroupId: string;
+
+  /**
+   * The workgroup endpoint address
+   */
+  readonly endpointAddress: string;
+
+  /**
+   * The workgroup port
+   */
+  readonly port: number;
+
+  /**
+   * The connection object associated with the Redshift Serverless Workgroup.
+   */
+  public readonly connections: aws_ec2.Connections;
+
+  private readonly props: WorkgroupProps;
+  private readonly securityGroups: aws_ec2.ISecurityGroup[];
+  private readonly vpcSubnets: aws_ec2.SubnetSelection;
+
+  constructor(scope: Construct, id: string, props: WorkgroupProps) {
+    super(scope, id, {
+      physicalName:
+        props.workgroupName ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 64, allowedSpecialCharacters: '-' }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+
+    this.securityGroups = props.securityGroups ?? [this.createSecurityGroup()];
+
+    this.connections = this.createConnections();
+
+    this.vpcSubnets = props.vpcSubnets ?? {
+      subnetType: aws_ec2.SubnetType.PRIVATE_WITH_EGRESS,
+    };
+
+    this.validateCapacity();
+    this.validateWorkgroupName();
+    this.validatePort();
+    this.validateSubnet();
+
+    const workgroup = this.createWorkgroup();
+
+    this.workgroupArn = workgroup.attrWorkgroupWorkgroupArn;
+    this.workgroupName = workgroup.attrWorkgroupWorkgroupName;
+    this.workgroupId = workgroup.attrWorkgroupWorkgroupId;
+    this.endpointAddress = workgroup.attrWorkgroupEndpointAddress;
+    this.port = workgroup.attrWorkgroupEndpointPort;
+  }
+
+  protected createWorkgroup(): aws_redshiftserverless.CfnWorkgroup {
```

Comment:
see https://github.com/open-constructs/aws-cdk-library/blob/main/src/aws-cur/cost-report.ts#L203

Reply:
same the above.